noobaa_sa/s3_client.py

```diff
@@ -370,6 +371,52 @@ def delete_bucket_policy(self, bucket_name):
             "delete_bucket_policy", Bucket=bucket_name
         )
         return response_dict
+    
+    def put_bucket_versioning(self, bucket_name, status="Enabled"):
+        """
+        Set versioning on bucket using boto3
+
+        Args:
+            Bucket_name (str): The name of the bucket
+            status (str): Versioning status
+        
+        Returns:
+            dict : PutBucketVersioning response
+        """
+        log.info(f"Enabling versioning status {status} on bucket {bucket_name} via boto3")
+        return self._exec_boto3_method(
+            "put_bucket_versioning", Bucket=bucket_name, VersioningConfiguration={"Status": status}
+        )
+    
+    def get_bucket_versioning(self, bucket_name):
+        """
+        Get versioning status of the bucket using boto3
+
+        Args:
+            Bucket_name (str): The name of the bucket
+        
+        Returns:
+            dict : GetBucketVersioning response
+        """
+        log.info(f"Getting versioning status of bucket {bucket_name} via boto3")
+        return self._exec_boto3_method(
+            "get_bucket_versioning", Bucket=bucket_name
+        )
+
+    def list_object_versions(self, bucket_name, **kwargs):
+        """
+        List all object versions present in bucket using boto3
+
+        Args:
+            Bucket_name (str): The name of the bucket
+        
+        Returns:
+            dict : list_object_versions response
+        """
+        log.info(f"List all object versions present in bucket {bucket_name} via boto3")
+        return self._exec_boto3_method(
+            "list_object_versions", Bucket=bucket_name, **kwargs
+        )
```

Comment:
Since we must pass an object key for this method to work - please add an explicit argument. It would enforce proper usage and add clarity on how to use the method.

Reply:
There is no need to pass object key and that is not mandatory while calling `list_object_versions` API. 
Referance link: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_object_versions.html